demos/markdown/tutorial/tutorial.didact.md

```diff
@@ -0,0 +1,129 @@
+
+```
+[Send some fantastic text to a Terminal window!](didact://?commandId=vscode.didact.sendNamedTerminalAString&text=TerminalName$$echo+Didact+is+fantastic%21)
+```
+
+Try it and if you get stuck, [click here](didact://?commandId=vscode.didact.copyToClipboardCommand&text=%5BSend%20some%20fantastic%20text%20to%20a%20Terminal%20window%21%5D%28didact%3A%2F%2F%3FcommandId%3Dvscode.didact.sendNamedTerminalAString%26text%3DTerminalName%2524%2524echo%2BDidact%2Bis%2Bfantastic%2521%29) to put the text above on the clipboard and paste it into your Markdown file.
+
+![Adding Requirements Checking](images/requirements-checking-header.png){.imageCenter}
+
+## Requirements Checking
+
+In order to use commands at the command line, often you want to ensure that the user is primed for success. To do that, we can actually check their system to make sure things are installed before they get going.
+
+For example:
+
+* Maybe we want to make sure the user has a particular VS Code extension pre-installed… 
+* Maybe we want to make sure the user has the Camel K or OpenShift CLI tools installed and ready…
+* Maybe we want to check to see if a component is set up in the user’s kubernetes cluster…
+
+Whatever we're checking, Didact likely has a way to make it work. And there's a two-step process:
+
+1. Create a label for the requirements result pass/fail
+2. Do the check and tie it to the label
+
+### Adding a Requirements Label
+
+Essentially the label is a textual placeholder for a success or failure message based on the requirement status. We have another type-ahead helper to help you create one.
+
+* Ctrl+Space, select `Insert Requirements label` and then make the hashtag label specific for your requirement.
+* Example: `*Status: unknown*{#requirement-name}`
+
+Then you have to decide what kind of requirement it is you’re checking and how specific you want to get.
+
+### Command-line Resource Checking
+
+If it’s a really simple command-line check and you just want to make sure it executes without errors (return code 0), you need the requirement label and the command line to try with `vscode.didact.cliCommandSuccessful`.
+
+* Ctrl+Space, select `Start a new Didact link`, and `Check CLI for Success (No Text)`, update the requirements label to use and the URL-encoded CLI text
+* Example to see if `mvn` is available in the system path: `didact://?commandId=vscode.didact.cliCommandSuccessful&text=maven-requirements-status$$mvn%20--version`
+
+Or you could do a more complex check and actually compare the first bit of text returned from the command execution with `vscode.didact.requirementCheck`.
+
+* Ctrl+Space, select `Start a new Didact link`, and `Check CLI for Returned Text`, update the requirements label to use, the URL-encoded CLI text, and whatever URL-encoded text you want to validate
+* Example that checks the text returned: `didact://?commandId=vscode.didact.requirementCheck&text=maven-requirements-status$$mvn%20--version$$Apache%20Maven`
```

Comment:
I had hard time to figure out what is the correct syntax:

I tried:
everythig stuck:
`*Status: unknown*{#demo-requirement}didact://?commandId=vscode.didact.requirementCheck&text=demo-requirement$$mvn%20--version$$Apache%20Maven`

then on a newline:
```
*Status: unknown*{#demo-requirement}
didact://?commandId=vscode.didact.requirementCheck&text=demo-requirement$$mvn%20--version$$Apache%20Maven
```
then with an empty line:
```
*Status: unknown*{#demo-requirement}

didact://?commandId=vscode.didact.requirementCheck&text=demo-requirement$$mvn%20--version$$Apache%20Maven
```

and finally:

```
*Status: unknown*{#demo-requirement}

[test requirement](didact://?commandId=vscode.didact.requirementCheck&text=demo-requirement$$mvn%20--version$$Apache%20Maven)
```

I didn't get that the `Status` is just a renderer of the result of another command in the file. I thought that the command was part of the `Status`

not sure how to improve that.
Maybe start by instructing user to provide a command with a link and then say that the status of it can be rendered using `*Status: unknown{#demo-requirement}`

Reply:
I'm confused with your confusion. :) I think it's pretty clear that the status area is created ahead of time. Would it be more helpful to simplify to one idea, then just get rid of this entire section around adding more requirements checking and directing them elsewhere?

Reply:
I like how you modify it to provide the full example first.